libethereum/ValidationSchemes.cpp

```diff
@@ -0,0 +1,142 @@
+        {{c_sealEngine, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_params, {{json_spirit::obj_type}, JsonFieldPresence::Required}},
+            {c_genesis, {{json_spirit::obj_type}, JsonFieldPresence::Required}},
+            {c_accounts, {{json_spirit::obj_type}, JsonFieldPresence::Required}}});
+
+    requireJsonFields(_obj.at(c_genesis).get_obj(), "ChainParams::loadConfig::genesis",
+        {{c_author, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_nonce, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_gasLimit, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_timestamp, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_difficulty, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_extraData, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_mixHash, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_parentHash, {{json_spirit::str_type}, JsonFieldPresence::Optional}}});
+
+    js::mObject const& accounts = _obj.at(c_accounts).get_obj();
+    for (auto const& acc : accounts)
+        validateAccountObj(acc.second.get_obj());
+}
+
+void validateAccountMapObj(js::mObject const& _obj)
+{
+    // A map object with possibly defined fields
+    requireJsonFields(_obj, "validateAccountMapObj",
+        {{c_storage, {{json_spirit::obj_type}, JsonFieldPresence::Optional}},
+            {c_balance, {{json_spirit::str_type}, JsonFieldPresence::Optional}},
+            {c_nonce, {{json_spirit::str_type}, JsonFieldPresence::Optional}},
+            {c_code, {{json_spirit::str_type}, JsonFieldPresence::Optional}},
+            {c_precompiled, {{json_spirit::obj_type}, JsonFieldPresence::Optional}},
+            {c_shouldnotexist, {{json_spirit::str_type}, JsonFieldPresence::Optional}},
+            {c_wei, {{json_spirit::str_type}, JsonFieldPresence::Optional}}});
+    }
+
+    void validateAccountObj(js::mObject const& _obj)
+    {
+        if (_obj.count(c_precompiled))
+        {
+            // A precompiled contract
+            requireJsonFields(_obj, "validateAccountObj",
+                {{c_precompiled, {{json_spirit::obj_type}, JsonFieldPresence::Required}},
+                    {c_wei, {{json_spirit::str_type}, JsonFieldPresence::Optional}}});
+        }
+        else if (_obj.size() == 1)
+        {
+            // A genesis account with only balance set
+            if (_obj.count(c_balance))
+                requireJsonFields(_obj, "validateAccountObj",
+                    {{c_balance, {{json_spirit::str_type}, JsonFieldPresence::Required}}});
+            else
+                requireJsonFields(_obj, "validateAccountObj",
+                    {{c_wei, {{json_spirit::str_type}, JsonFieldPresence::Required}}});
+        }
+        else
+        {
+            // A standart account with all fields
```

Comment:
What's the difference between standard account and genesis account? Do these standard accounts really always have code and storage?

Reply:
yes. standart account required to have all 4 fields. 
in genesis it is not required.
and account map could be empty